provider/provider.go

```diff
@@ -325,6 +334,18 @@ func (p *dockerNativeProvider) Create(ctx context.Context, req *rpc.CreateReques
 		return nil, errors.Wrapf(err, "malformed resource inputs")
 	}
 
+	// if we're in preview mode and buildOnPreview is set to false, we return the inputs
+	if req.GetPreview() && !inputs["buildOnPreview"].BoolValue() {
```

Comment:
What if `inputs["buildOnPreview"]` is unknown?

1. This code will panic.
2. What should the provider do?

Reply:
Again, thinking about this from a user perspective, I was having a difficult time coming up with a situation where anyone would make buildOnPreview unknown, but you're right that the world is a large and diverse place so we should add a check. My feeling is to warn and continue.